state/protocol/protocol_state/kvstore/models.go

```diff
@@ -67,25 +68,29 @@ func (model *Modelv0) ID() flow.Identifier {
 	return makeVersionedModelID(model)
 }
 
-// Replicate instantiates a Protocol State Snapshot of the given `protocolVersion`.
-// It clones existing snapshot and performs a migration if `protocolVersion = 1`.
+// Replicate instantiates a Protocol State Snapshot of the given protocolVersion.
+// It clones existing snapshot if protocolVersion = currentVersion.
+// It transitions to next version if protocolVersion = currentVersion+1.
 // Expected errors during normal operations:
 //   - ErrIncompatibleVersionChange if replicating the Parent Snapshot into a Snapshot
 //     with the specified `protocolVersion` is not supported.
 func (model *Modelv0) Replicate(protocolVersion uint64) (protocol_state.KVStoreMutator, error) {
-	version := model.GetProtocolStateVersion()
-	if version == protocolVersion {
+	currentVersion := model.GetProtocolStateVersion()
+	if currentVersion == protocolVersion {
 		// no need for migration, return a complete copy
 		return clone.Clone(model), nil
 	} else if protocolVersion != 1 {
-		return nil, fmt.Errorf("unsupported replication version %d, expect %d: %w",
+		return nil, fmt.Errorf("unsupported replication currentVersion %d, expect %d: %w",
 			protocolVersion, 1, ErrIncompatibleVersionChange)
 	}
 
-	// perform actual replication to the next version
+	// perform actual replication to the next currentVersion
```

Comment:
I think this change is incorrect. `currentVersion` should always be `0`, because it is the version of the parent model: 
https://github.com/onflow/flow-go/blob/742d7005009f2f640f816b4de4b718fd7b887c4e/state/protocol/protocol_state/kvstore/models.go#L78

I think if you accepted [my comment above](), we could just skip this here and the code would read

```golang
// version change: Modelv0 only supports upgrade to protocolVersion = 1
if protocolVersion != 1 {
  return nil, fmt.Errorf("protocol state's current version %d only supports replication into version %d but requested was version %d: %w",
    currentVersion, 1, protocolVersion, ErrIncompatibleVersionChange)
}
v1 := &Modelv1{
  Modelv0: clone.Clone(*model),
}
```

Reply:
Like the above, the `currentVersion` in the error message is just a refactor error (fixed now).